### PR TITLE
Feature/styles

### PR DESCRIPTION
**Pull request by coditive-tomek-g**
**Added:** 
- Basic styles structure
- Basic scripts structure - @coditive-przemek-h - please take a look
- Post tile component
- Stylelint config
- Theme.json settings (typography, colors, font-sizes)

### CODE REVIEW

File: resources/views/index.blade.php [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
+      @endwhile
+    </div>
 
-    {!! get_search_form(false) !!}
+    <div class="pagination">
+      {!! get_the_posts_navigation() !!}
+    </div>
+  @else
+    <div class="error-message">
+      {{ __('Sorry, no results were found.', 'wps') }}
```
Comment (coditive-pawel-m) [inline resources/views/index.blade.php:21]: @coditive-tomek-g @coditive-przemek-h  I assume that we will change `wps` to `firestarter`?

File: app/View/Components/PostTile.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\View\Components;
+
+use Illuminate\View\Component;
+
+class PostTile extends Component
+{
+  public $class;
+  public $title;
```
Comment (coditive-przemek-h) [inline app/View/Components/PostTile.php:10]: Can you add a new lines between attributes?

File: resources/views/sections/site-footer.blade.php [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<footer class="site-footer -wrapper">
+  <div class="site-footer__widgets">
+    @php(dynamic_sidebar('sidebar-footer'))
+  </div>
+
+  <p class="site-footer__copyrights">
+    To do: Add copyrights text from ACF options
```
Comment (coditive-pawel-m) [inline resources/views/sections/site-footer.blade.php:7]: For todo's please use this convention:
html:
`<!-- TODO: some todo -->`

js:
`// TODO: some todo`

So we can have syntax highlighting while using vs code plugins like Better Comments or create issues directly through ide (I recommend GitHub Pull Requests and Issues for vs code - ratings are poor but it works for me ;))

cc: @coditive-przemek-h

**Review by coditive-pawel-m — CHANGES_REQUESTED** [review]
Add some small comments

File: app/View/Components/PostTile.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  public function __construct($class, $postId = null)
+  {
+    $this->class = $class;
+    $this->title = $this->getTitle();
+    $this->link = $this->getLink();
+    $this->excerpt = $this->getExcerpt();
+  }
+
+  protected function getTitle()
```
Comment (coditive-przemek-h) [inline app/View/Components/PostTile.php:29]: I think that `getters` should return attribute values in this case. It seems to be ok if you would use `getTitle` function in the components, but in this case, getters are used only for setting attributes, so the meaning might be confusing. To I would use setters in constructors instead.

```php
public function __construct($class, $postId = null)
{
    $this->setTitle(get_the_title($this->postId));
}

private function setTitle(string $title): void
{
  $this->title = $title;
}
```

File: app/View/Composers/App.php [hunk 5]
```diff
@@ -20,20 +20,32 @@ class App extends Composer
...
-    /**
-     * Returns the site name.
-     *
-     * @return string
-     */
-    public function siteName()
-    {
-        return get_bloginfo('name', 'display');
+
+  public function siteName(): string
```
Comment (coditive-przemek-h) [inline app/View/Composers/App.php:31]: You return logo in `siteName` function?

File: resources/styles/abstracts/_variables.scss [hunk 6]
```diff
@@ -0,0 +1,7 @@
+:root {
+  /**
```
Comment (coditive-przemek-h) [inline resources/styles/abstracts/_variables.scss:2]: It seems that incorrect formatting is placed here. Should we keep comments in the code?

File: app/View/Composers/App.php [hunk 7]
```diff
@@ -20,20 +20,32 @@ class App extends Composer
...
+      if (empty($logoStandard)) {
+        return get_bloginfo('name');
+      }
+
+      $srcset = '';
+      if (! empty($logoStandard) && ! empty($logoRetina)) {
+        $srcset = 'srcset="' . $logoStandard . ' 1x, ' . $logoRetina . ' 2x"';
+      }
+
+      return '<img src="' . $logoStandard . '" ' . $srcset . ' />';
```
[the remark this answers is not in the document]
Comment (coditive-tomek-g) [reply app/View/Composers/App.php:46]: @coditive-pawel-m - sure, but currently we don't have ACF settings added to the theme, so I'll do it when @coditive-przemek-h adds it.

File: resources/views/components/post-tile.blade.php [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<article class="{{ $class }} post-tile">
+  <header class="post-tile__header">
+    <h4 class="post-tile__heading">
+      <a href="{{ $link }}" class="post-tile__link">
+        {{ $title }}
+      </a>
+    </h4>
+  </header>
+
+  <p class="post-tile__excerpt">
```
Comment (coditive-przemek-h) [inline resources/views/components/post-tile.blade.php:10]: Sometimes excerpts might not exist so we will keep empty paragraphs in such cases. Shouldn't we show excerpt only when exists?

File: resources/views/index.blade.php (hunk 1 above)
Comment (coditive-tomek-g) [reply resources/views/index.blade.php:21]: @coditive-pawel-m - I think we should use `sage` as a text-domain, @coditive-przemek-h do you agree? If yes I'll change it.

**Review by coditive-przemek-h — CHANGES_REQUESTED** [review]
First review has been made.

File: app/View/Components/PostTile.php [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+class PostTile extends Component
+{
+  public $class;
+  public $title;
+  public $link;
+  public $excerpt;
+  protected $postId;
+
+  /**
```
[the remark this answers is not in the document]
Comment (coditive-tomek-g) [reply app/View/Components/PostTile.php:15]: @coditive-przemek-h - These comments are auto-generated during using `wp acorn make:component ExampleComponent` command, but of course, I can remove them.

File: resources/styles/abstracts/_variables.scss (hunk 6 above)
Comment (coditive-tomek-g) [reply resources/styles/abstracts/_variables.scss:2]: I think we can keep comments in such a "config" scss files to leave information about the mixins, variables etc. What do you think?

**Review by coditive-przemek-h — APPROVED** [review]
Please add also type for `protected $postId` and thats all :D

**Pull request merged**